app/src/main/java/com/github/featuredetectandroid/MainActivity.kt

```diff
@@ -35,6 +40,7 @@ private const val RESOLUTION_HEIGHT = 360
 class MainActivity : ComponentActivity() {
     private val cameraExecutor = Executors.newSingleThreadExecutor()
     private val imageViewModel by viewModels<GrayscaleViewModel>()
+    private lateinit var currentAlgorithm: SharedPreferences
```

Comment:
It's wrong naming as it's not an algorithm -- it's just something where you retrive algorithm name from. Name it something like `preferences`.

Reply:
fixed